compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -314,6 +314,9 @@ impl<'a> Linker for GccLinker<'a> {
                 self.cmd.arg("-static");
                 self.build_dylib(out_filename);
             }
+            LinkOutputKind::WasiReactorExe => {
+                panic!("can't link as reactor on non-wasi target");
```

Comment:
I think using `GccLinker` is possible if you're compiling wasi with a native clang toolchain locally? I think though the fix for this would just be some `-Wl,` arguments though?

Reply:
I think I can just use `--entry _initialize` then, if that's a case that can happen? It looks like both ld and lld (or gcc and wasmld, whatever) support the same `--entry foo` syntax, so hopefully just adding that would work.